subworkflows/local/input_check.nf

```diff
@@ -10,21 +10,45 @@ workflow INPUT_CHECK {
     samplesheet // file: /path/to/samplesheet.csv
 
     main:
-    SAMPLESHEET_CHECK ( samplesheet )
-        .csv
-        .splitCsv ( header:true, sep:',' )
-        .map { create_fastq_channel(it) }
-        .groupTuple(by: [0]) // group replicate files together, modifies channel to [ val(meta), [ [reads_rep1], [reads_repN] ] ]
-        .map { meta, reads -> [ meta, reads.flatten() ] } // needs to flatten due to last "groupTuple", so we now have reads as a single array as expected by nf-core modules: [ val(meta), [ reads ] ]
-        .set { reads }
+
+    reads = null
+    versions = null
+
+    if (params.aligner == "cellrangerarc"){
+        SAMPLESHEET_CHECK ( samplesheet )
```

Comment:
Since the first part is the same, I would suggest **something like** this to avoid code repetition and be prone to typos:

```groovy
grouped_ch =
SAMPLESHEET_CHECK ( samplesheet )
      .csv
      .splitCsv ( header:true, sep:',' )
      .map { create_fastq_channel(it) }
      // group replicate files together, modifies channel to [ val(meta), [ [reads_rep1], [reads_repN] ] ]
      .groupTuple(by: [0])

if (params.aligner == 'cellrangerarc' ) {
    grouped_ch
    .map { meta, sample_type, sub_sample, reads -> [ meta, sample_type.flatten(), sub_sample.flatten(), reads.flatten() ] } 
    .set { reads }
} else {
    grouped_ch
    .map { meta, reads -> [ meta, reads.flatten() ] } 
    .set { reads }
}

versions = SAMPLESHEET_CHECK.out.versions
```